pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -3265,6 +3284,48 @@ func createFederationRelationship(tx *gorm.DB, fr *datastore.FederationRelations
 	return fr, nil
 }
 
+func fetchFederationRelationship(tx *gorm.DB, trustdomain spiffeid.TrustDomain) (*datastore.FederationRelationship, error) {
+	var model FederatedTrustDomain
+	err := tx.Find(&model, "trust_domain = ?", trustdomain.String()).Error
+	switch {
+	case errors.Is(err, gorm.ErrRecordNotFound):
+		return nil, nil
+	case err != nil:
+		return nil, sqlError.Wrap(err)
+	}
+
+	bundleEndpointURL, err := url.Parse(model.BundleEndpointURL)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse URL: %w", err)
+	}
+
+	fr := &datastore.FederationRelationship{
+		TrustDomain:           trustdomain,
+		BundleEndpointURL:     bundleEndpointURL,
+		BundleEndpointProfile: datastore.BundleEndpointType(model.BundleEndpointProfile),
+	}
+
+	switch fr.BundleEndpointProfile {
+	case datastore.BundleEndpointWeb:
+	case datastore.BundleEndpointSPIFFE:
+		endpointSPIFFEID, err := spiffeid.FromString(model.EndpointSPIFFEID)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse bundle endpoint SPIFFE ID: %w", err)
+		}
+		fr.EndpointSPIFFEID = endpointSPIFFEID
+
+		bundle, err := fetchBundle(tx, trustdomain.IDString())
```

Comment:
@amartinezfayo , @evan2645, I'm curious what you think of this. Do we expect folks who configure a relationship with a SPIFFE profile to see the bundle they configured or the latest bundle when they fetch?

Reply:
I _feel_ like the natural thing is for this to be the latest bundle? Especially if folks want to copy/paste or move things around to other clusters and have it work.

It was also my understanding that when a relationship is created and a bundle is explicitly supplied, that we would simply write this bundle to the bundles table and use it to bootstrap as we normally would (as opposed to storing the originally configured bundle with the relationship). If we were doing that, I think the "latest" bundle is all we could afford anyway?

Reply:
You're right. I concur. Ship it!

Reply:
Yes, I also think that he natural behavior is to be the latest bundle.